### PR TITLE
ci: attempt to increase stability of saucelabs legacy job

### PR DESCRIPTION
**Pull request by devversion**
See individual commits

### CODE REVIEW

File: karma-js.conf.js [hunk 1]
```diff
@@ -151,6 +151,15 @@ module.exports = function(config) {
     set: () => {},
   });
 
+  if (process.env.CIRCLECI) {
+    conf.plugins.push(require('karma-parallel'));
+    conf.frameworks.push('parallel');
```
Comment (gkalpak) [inline karma-js.conf.js:156]: The docs say this has to be the first, so you might need to `unshift` instead 🧐

File: packages/core/test/render3/providers_helper.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+          ]
+    });
+  }
+
+
+  const fixture = new ComponentFixture(
+      App, {injector: defs.ngModule ? createInjector(defs.ngModule) : undefined});
+  expect(fixture.html).toEqual('<parent><view-child>view-child</view-child></parent>');
+
+  fixture.destroy();
```
Comment (devversion) [inline packages/core/test/render3/providers_helper.ts:177]: This helper has just been moved from the spec file to a shared helper file. The only addition is this `fixture.destroy` call here; allowing us to also test that destroy hooks are invoked properly.

File: packages/compiler-cli/src/transformers/downlevel_decorators_transform/downlevel_decorators_transform.ts [hunk 3]
```diff
@@ -465,9 +467,8 @@ export function getDownlevelDecoratorsTransform(
             param.dotDotDotToken, param.name, param.questionToken, param.type, param.initializer);
         newParameters.push(newParam);
       }
-      const updated = ts.updateConstructor(
-          ctor, ctor.decorators, ctor.modifiers, newParameters,
-          ts.visitFunctionBody(ctor.body, decoratorDownlevelVisitor, context));
+      const updated =
```
Comment (AndrewKushnir) [inline packages/compiler-cli/src/transformers/downlevel_decorators_transform/downlevel_decorators_transform.ts:470]: @alxhub could you please help with the review of the compiler-related changes?

File: packages/core/test/render3/providers_es2015_spec.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {forwardRef, InjectionToken, ɵɵdirectiveInject} from '@angular/core';
+
+import {expectProvidersScenario} from './providers_helper';
+
+describe('es2015 providers', () => {
```
Comment (AndrewKushnir) [inline packages/core/test/render3/providers_es2015_spec.ts:13]: It'd be great to add a description on why we have special tests for providers + ES2015 (+ similar comment in the packages/core/test/render3/BUILD.bazel file as well).

File: packages/core/test/render3/providers_helper.ts (hunk 2 above)
Comment (AndrewKushnir) [reply packages/core/test/render3/providers_helper.ts:177]: I'm wondering if we can have an acceptance test instead (the one in the `packages/core/test/acceptance` folder) vs creating a new test that would use hand-written generated code (which we try to avoid in general)?

File: packages/router/karma.conf.js [hunk 5]
```diff
@@ -1,113 +0,0 @@
-/**
```
Comment (AndrewKushnir) [inline packages/router/karma.conf.js:1]: Awesome that we don't have a special Karma config here 👍

**Review by AndrewKushnir — APPROVED** [review]
@devversion thanks for fixing the issues and making CI more stable 👍 

The changes look good (just a couple comments), but I'd like to ask @alxhub to take a look at the compiler-related changes as well (related to the downlevelling logic).

**Review by josephperrott — APPROVED** [review]
LGTM

Reviewed-for: dev-infra

**Review by atscott — APPROVED** [review]

File: packages/core/test/render3/providers_es2015_spec.ts (hunk 4 above)
Comment (AndrewKushnir) [reply packages/core/test/render3/providers_es2015_spec.ts:13]: FYI, I'm ok to get my feedback addressed in a followup PR, plz don't block the submission on my feedback - there is nothing critical.

File: karma-js.conf.js [hunk 6]
```diff
@@ -151,6 +149,15 @@ module.exports = function(config) {
     set: () => {},
   });
 
+  if (process.env.CIRCLECI) {
```
Comment (gkalpak) [inline karma-js.conf.js]: Nit: Why not `process.env.CI` (which is what we typically use)?

File: packages/core/test/animation/animations_with_web_animations_integration_spec.ts [hunk 7]
```diff
@@ -249,7 +249,8 @@ describe('animation integration tests using web animations', function() {
                 overflow:hidden;
               }
               .list .inner {
-                line-height:50px;
+                box-sizing: border-box;
+                height: 50px;
```
Comment (gkalpak) [inline packages/core/test/animation/animations_with_web_animations_integration_spec.ts:253]: OOC, why this change?

**Review by gkalpak — APPROVED** [review]

File: karma-js.conf.js (hunk 6 above)
Comment (gkalpak) [reply karma-js.conf.js]: Nevermind, this code was removed in the next commit.

Comment (devversion) [reply karma-js.conf.js]: I should have cleaned this up but didn't realize. This PR really grew too much with all the other necessary fixes, sorry! 😇😢

File: packages/core/test/animation/animations_with_web_animations_integration_spec.ts (hunk 7 above)
Comment (devversion) [reply packages/core/test/animation/animations_with_web_animations_integration_spec.ts:253]: line height behaved flaky in Android 10, and this fixed it

File: karma-js.conf.js (hunk 6 above)
Comment (devversion) [reply karma-js.conf.js]: I will try to fix this if the PR does not land today.

File: packages/core/test/render3/providers_es2015_spec.ts (hunk 4 above)
Comment (devversion) [reply packages/core/test/render3/providers_es2015_spec.ts:13]: As discussed offline: Definitely agreed. Also on it being an acceptance test instead.

If this PR does not make it today, I'll fix this up tomorrow.

**Comment by dylhunn** [comment]
Igor is the only reviewer on fw-benchmarks and is no longer with us, so we can ignore pullapprove bot for now. What target label does this need?

**Review by alxhub — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
Merge-assistance: this PR is ready for merge. Alex's approval covers the remaining `fw-benchmarks` group as well (Alex will be a part of that group once PR #44304 is merged).

File: packages/core/test/render3/providers_es2015_spec.ts (hunk 4 above)
Comment (AndrewKushnir) [reply packages/core/test/render3/providers_es2015_spec.ts:13]: @devversion since this change was presubmitted in g3, let's keep everything as is and create a followup PR (so that we don't need to re-run a presubmit)?

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 0fcf18d5f476cb2be6c89d28185fdf8ab34251f4.
